src/commands/git-metadata/library.ts

```diff
@@ -0,0 +1,85 @@
+import {newApiKeyValidator} from '../../helpers/apikey'
+import {RequestBuilder} from '../../helpers/interfaces'
+import {upload, UploadOptions, UploadStatus} from '../../helpers/upload'
+import {getRequestBuilder} from '../../helpers/utils'
+import {getCommitInfoBasic, newSimpleGitOrFail} from './git'
+import {CommitInfo} from './interfaces'
+
+export class SourceCodeIntegration {
+  private apiKey: string
+  private datadogSite: string
+  private version = require('../../../package.json').version
+
+  constructor(apiKey: string, datadogSite?: string) {
+    this.datadogSite = datadogSite ?? 'datadoghq.com'
+    this.apiKey = apiKey
+  }
+
+  public static async shouldAddSourceCodeIntegration(apiKey: string | undefined): Promise<boolean> {
+    let simpleGit
+    let isRepo
+    try {
+      simpleGit = await newSimpleGitOrFail()
+      isRepo = simpleGit.checkIsRepo()
+    } catch {
+      return false
+    }
+
+    // Only enable if the system has `git` installed and we're in a git repo
+    return apiKey !== undefined && isRepo
+  }
+
+  public async addSourceCodeIntegration(): Promise<string> {
```

Comment:
Should this be renamed to something like `uploadGitMetadata` since `add` doesn't really describe what's happening under-the-hood? Just want to make sure since it'll be imported elsewhere.

Reply:
Good catch! I think `uploadGitCommitHash` is descriptive enough.